OrbitBase/MainThreadExecutor.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitBase/MainThreadExecutor.h"
+
+#include <list>
+#include <thread>
+
+#include "OrbitBase/Logging.h"
+#include "absl/synchronization/mutex.h"
+
+namespace {
+
+class MainThreadExecutorImpl : public MainThreadExecutor {
+ public:
+  explicit MainThreadExecutorImpl(std::thread::id main_thread_id)
+      : main_thread_id_(main_thread_id) {}
+
+  void Schedule(std::shared_ptr<Action> action) override;
+
+  void ConsumeActions() override;
+
+ private:
+  std::shared_ptr<Action> PopAction();
+
+  std::thread::id main_thread_id_;
+  absl::Mutex mutex_;
+  std::list<std::shared_ptr<Action>> scheduled_actions_;
+};
+
+void MainThreadExecutorImpl::Schedule(std::shared_ptr<Action> action) {
```

Comment:
In think that the pointer can be passed as const ref or `std::move`d on `push_back`.

Reply:
replaced with unique_ptr